hashintel/hash
# Add tooltip support to entity editor

## 🌟 What is the purpose of this PR?

This PR adds tooltip support on glide-grid cells. This feature will be used for UX purposes like  indicating which property is required in the table, or showing the full text in a cell, if the text is overflowing. 


## 🔗 Related links

- [Asana task](https://app.asana.com/0/1203007126736607/1203191162200434/f) _(internal)_
- [Design](https://www.figma.com/file/gydVGka9FjNEg9E2STwhi2/HASH-App?node-id=4968%3A243891) _(internal)_

## 🔍 What does this change?

- Adds `customGridIcons`, so we can render those icons via `SpriteRenderer` of `glide-data-grid`
- Refactor `table sorting` codes based on new types
- Create `useGridTooltip` hook, and use it with `properties-table`
- Create custom `value-cell` to handle property values
  - This handles only text values for now

## Known Issues

See next-steps below

## 🐾 Next steps

- Property value tooltips for error state
- Property value tooltips for label property
- Make tooltips available on mobile devices
- Prevent tooltip popping-up while user is editing the cell
- Improve tooltip-hiding logic

## ❓ How to test this?

1.  Checkout the branch / view the deployment
1.  Login
2. Copy your entity-id you get from `http://localhost:3000/entity-editor`
3. Open `http://localhost:3000/@{shortname}/entities/{userId}`  using the entity-id you copied on previous step
1.  Confirm that tooltips are working as expected

## 📹 Demo

<img width="794" alt="Screen Shot 2022-10-24 at 18 11 33" src="https://user-images.githubusercontent.com/39553853/197561241-bd7588f4-7002-4a07-94b8-662fc017edb1.png">

https://user-images.githubusercontent.com/39553853/197561062-511aedf1-f5bb-4261-b91f-81a5a7d7227c.mp4
